Add an API method to switch controller for agent

dofmind opened the pull request:
This PR is related with #819.

This adds an API method to set the ControllerAddress= which triggers a disconnect in the agent, and a property to emits a signal that the address has been changed.

Also fixes errors detected by Lint to check apispec.

engelmi commented on src/agent/agent.c:
```diff
@@ -1887,6 +1942,11 @@ static const sd_bus_vtable agent_vtable[] = {
+                        "s",
+                        agent_property_get_controller_address,
+                        0,
+                        SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```
```suggestion
        SD_BUS_PROPERTY("ControllerAddress",
                        "s",
                        NULL,
                        offsetof(Agent, orch_addr),
                        SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```

Instead of implementing a dedicated function we can use `offsetof` since the property is a field on the `Agent` struct. So `agent_property_get_controller_address` can be removed. 

Also, I'd use `agent->orch_addr` instead of `agent->controller_address` for the property. The "final" address of the controller where the agent connects to is assembled either from
- `host` and `port`
- or `controller_address`

and saved to `agent->orch_addr`. So by using this field, the property shows the actual address used.

dofmind replied:
Thanks for the feedback, i updated.

engelmi replied:
Thanks, looks good!

engelmi reviewed (commented):
Thank you for your contribution! @dofmind 
I added one comment, otherwise the PR looks good to me.

42dot-joonyoung-shim committed:
Add an API method to switch controller for agent

The bluechi-controller is currently a single point of failure. In order
to address this issue and increase the resilience of the system,
bluechi-agent should provide an API method to set the ControllerAddress=
which triggers a disconnect in the agent and emits a signal that the
address has been changed.

Signed-off-by: Joonyoung Shim <joonyoung.shim@42dot.ai>

42dot-joonyoung-shim committed:
Fix changes in generated python code

This fixes errors detected by Lint to check apispec.

Signed-off-by: Joonyoung Shim <joonyoung.shim@42dot.ai>

alexlarsson commented on src/agent/agent.c:
```diff
@@ -1808,6 +1808,43 @@ static int agent_method_remove_proxy(sd_bus_message *m, UNUSED void *userdata, U
+                                strerror(-r));
+        }
+
+        if (!agent_set_controller_address(agent, dbus_address)) {
```
This only sets agent->controller_address, but if ensure_orch_address has already run, then this has been copied to agent->orch_addr, (and that is what the property reads)  and this newly set value will never be used. So, I can't see how this works at all? 

What you want here is agent_set_orch_address, surely?

alexlarsson replied:
Ah, i see that orch_addr is reset on reconnect, and we now always reconnect, so i guess this is fine.

engelmi reviewed (approved):
LGTM
